### PR TITLE
#87 : use same timeout as for underlying projection

### PR DESCRIPTION
**Pull request by cvanfleteren**
Use at least same timeout as underlying projection. Otherwise the projection will eventually succeed, but your command will have failed with a timeout when joining, since the EventsMonitorActor will never wait long enough to notify the AggregateActorRef with the Done message, even if you were willing to wait long enough when making the initial ask to the aggregate.

### CODE REVIEW

File: modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala [hunk 1]
```diff
@@ -144,7 +146,9 @@ class ProjectionMonitorActor extends Actor with ActorLogging {
         tryComplete()
 
       case ReceiveTimeout =>
-        log.debug("Got timeout")
+        log.debug(
+          s"Got timeout after waiting ${timeout.toSeconds}s for ${eventsToWaitFor.size} events. Received ${eventsReceived.size} in the meantime"
```
Comment (rcavalcanti) [inline modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala]: Much better logging. Thanks

File: modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala [hunk 2]
```diff
@@ -107,8 +108,9 @@ class ProjectionMonitorActor extends Actor with ActorLogging {
 
     val monitorName = s"monitor::$projectionName::${commandId.value}::${UUID.randomUUID()}"
 
-    // TODO this could be moved to property file eventually
-    val shutdownEventsMonitorAfter = 10.seconds
+    // take at least the same timeout as the underlying projection
+    // if you're willing to wait X seconds on the projection, you should be willing to wait X seconds for events coming from the Projection
+    val shutdownEventsMonitorAfter = projectionConfig(projectionName).getDuration("async-projection-timeout", 5.seconds) + 1.second
```
Comment (rcavalcanti) [inline modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala]: maybe instead `+ 1.second` we should have a `* factor`? What do you think?

Comment (rcavalcanti) [reply modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala]: That would required an extra param

Comment (cvanfleteren) [reply modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala]: Actually, thinking about it a bit more, maybe the "proper" value for this timeout should depend on the askTimeout you're using to talk to the AggregateActorRef, but that would require much more intrusive changes.

I'm willing to add the extra factor if you feel it's necessary. I don't think it's that useful to be honest.

Comment (rcavalcanti) [reply modules/akka/src/main/scala/io/funcqrs/akka/ProjectionMonitorActor.scala]: that's ok, let's leave it like that.

**Comment by rcavalcanti** [comment]
@cvanfleteren this was release today. Artifacts for v1.0.1 should show up soon in Maven Central.

**Pull request merged**